app/queries/content_dependencies.rb

```diff
@@ -30,16 +30,37 @@ def call
       with_locales - [calling_item]
     end
 
+    def priorities
+      dependencies = dependency_resolution.dependencies_with_priorities
+      dependencies.map do |dependency|
+        _, locale = Queries::LocalesForEditions.call([dependency[:content_id]], content_stores).first
```

Comment:
I think we should try avoid doing this in a map since that'll introduce a ton of extra queries. 

You probably want to get all content_ids and priorities grouped in a hash: `dependencies = { content_id_1 => priority, content_id_1 => priority }`
Then get all the locales: `locales = LocalesForEditions.call(dependencies.keys, content_stores)`
Then you can probably map the locale array: `locales.map { |(content_id, locale)| { content_id: content_id, locale: locale, priority: dependencies[content_id] } }`

Reply:
That makes sense. The way I had it, `dependencies` contained an array of `{ content_id: "content_id", priority: :priority }`